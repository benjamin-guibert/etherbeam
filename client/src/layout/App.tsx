import React, { FC, useEffect, useState } from 'react'
import { hot } from 'react-hot-loader'
import { BrowserRouter } from 'react-router-dom'
import { Switch, Route } from 'react-router-dom'
import { User } from '../libraries/types'
import { SessionContext, useSessionContextValue } from './SessionContext'
import { ToastType, useToaster } from './toaster-helper'
import Header from './Header'
import Toaster from './Toaster'
import TokensPageWrapper from './blockchain/TokensPageWrapper'
import TokenPageWrapper from './blockchain/TokenPageWrapper'
import './App.scss'
import { initializeServerData } from 'libraries/server'

const serverData = initializeServerData()

const App: FC = () => {
  const [toasts, setToast] = useState<ToastType[]>([])
  const { addToast } = useToaster(toasts, setToast)
  const sessionContextValue = useSessionContextValue(serverData)
  const { initialize } = sessionContextValue

  useEffect(() => {
    initialize()
  }, [initialize])

  return (
    <SessionContext.Provider value={sessionContextValue}>
      <BrowserRouter>
        <div className="my-app">
          <Header />
          <main className="my-app-main">
            <Switch>
              <Route path="/" exact></Route>
              <Route path="/tokens" exact>
                <TokensPageWrapper serverData={serverData} addToast={addToast} />
              </Route>
              <Route path="/tokens/:address" exact>
                <TokenPageWrapper serverData={serverData} addToast={addToast} />
              </Route>
            </Switch>
          </main>
          <Toaster toasts={toasts} />
        </div>
      </BrowserRouter>
    </SessionContext.Provider>
  )
}

export default hot(module)(App)
