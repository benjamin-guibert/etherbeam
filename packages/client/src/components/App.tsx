import React, { ReactElement, useState } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { hot } from 'react-hot-loader'
import { Switch, Route } from 'react-router-dom'
import Container from 'react-bootstrap/Container'
import Toaster, { ToastContent } from './Toaster'
import NavigationBar from './NavigationBar'
import TokensPage from './modules/blockchain/TokensPage'
import TokenPage from './modules/blockchain/TokenPage'

const App = (): ReactElement => {
  const [toasts, setToasts] = useState<ToastContent[]>([])

  const addToast = (toast: ToastContent) => {
    setToasts((oldToasts) => [toast, ...oldToasts].slice(0, 5))
  }

  return (
    <BrowserRouter>
      <div>
        <header>
          <NavigationBar />
        </header>
        <Container>
          <Switch>
            <Route path="/" exact></Route>
            <Route path="/tokens" exact>
              <TokensPage addToast={addToast} />
            </Route>
            <Route path="/tokens/:address" exact>
              <TokenPage addToast={addToast} />
            </Route>
          </Switch>
        </Container>
        <Toaster toasts={toasts} />
      </div>
    </BrowserRouter>
  )
}

export default hot(module)(App)
