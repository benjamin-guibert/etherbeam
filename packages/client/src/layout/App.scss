@use '../styles/resources/sizes';

.my-app {
  display: grid;
  grid-template-rows: auto auto auto;
  grid-template-areas:
    'header'
    'main'
    'footer';
}
