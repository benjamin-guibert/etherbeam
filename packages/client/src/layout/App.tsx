import React, { ReactElement } from 'react'
import { hot } from 'react-hot-loader'
import { BrowserRouter } from 'react-router-dom'
import { Switch, Route } from 'react-router-dom'
import Header from './Header'
import TokensPage from 'layout/blockchain/TokensPage'
import TokenPage from 'layout/blockchain/TokenPage'
import './App.scss'

const App = (): ReactElement => {
  return (
    <BrowserRouter>
      <div className="my-app">
        <Header />
        <Switch>
          <Route path="/" exact></Route>
          <Route path="/tokens" exact>
            <TokensPage />
          </Route>
          <Route path="/tokens/:address" exact>
            <TokenPage />
          </Route>
        </Switch>
      </div>
    </BrowserRouter>
  )
}

export default hot(module)(App)
