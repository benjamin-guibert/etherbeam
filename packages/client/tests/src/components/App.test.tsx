import * as React from 'react'
import { render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import App from 'components/App'

describe('<App />', () => {
  let component: unknown

  describe('Home page', () => {
    beforeAll(() => {
      component = render(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      )
    })

    it('should match snapshot', () => expect(component).toMatchSnapshot())
  })

  describe('Invalid path', () => {
    beforeAll(() => {
      component = render(
        <MemoryRouter initialEntries={['/invalid-path']}>
          <App />
        </MemoryRouter>
      )
    })

    it('should match snapshot', () => expect(component).toMatchSnapshot())
  })

  describe('Tokens', () => {
    beforeAll(() => {
      component = render(
        <MemoryRouter initialEntries={['/tokens']}>
          <App />
        </MemoryRouter>
      )
    })

    it('should match snapshot', () => expect(component).toMatchSnapshot())
  })

  describe('Token', () => {
    beforeAll(() => {
      component = render(
        <MemoryRouter initialEntries={['/tokens/0x0000000000000000000000000000000000000111']}>
          <App />
        </MemoryRouter>
      )
    })

    it('should match snapshot', () => expect(component).toMatchSnapshot())
  })
})
