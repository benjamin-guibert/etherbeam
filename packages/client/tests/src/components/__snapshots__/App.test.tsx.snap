// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Home page should match snapshot 1`] = `
<div>
  <header>
    <nav
      class="mb-4 navbar navbar-expand navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
        id="navbar-home"
      >
        <img
          alt=""
          class="app-icon app-icon-lg"
          src=""
        />
        Etherbeam
      </a>
      <button
        aria-controls="navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbar-nav"
      >
        <a
          class="nav-link"
          href="/tokens"
          id="navbar-nav-tokens"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-dot-circle fa-w-16 fa-lg mr-1"
            data-icon="dot-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
              fill="currentColor"
            />
          </svg>
          Tokens
        </a>
      </div>
    </nav>
  </header>
  <div
    class="container"
  />
  <div
    class="fixed-top m-3"
  />
</div>
`;

exports[`<App /> Invalid path should match snapshot 1`] = `
<div>
  <header>
    <nav
      class="mb-4 navbar navbar-expand navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
        id="navbar-home"
      >
        <img
          alt=""
          class="app-icon app-icon-lg"
          src=""
        />
        Etherbeam
      </a>
      <button
        aria-controls="navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbar-nav"
      >
        <a
          class="nav-link"
          href="/tokens"
          id="navbar-nav-tokens"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-dot-circle fa-w-16 fa-lg mr-1"
            data-icon="dot-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
              fill="currentColor"
            />
          </svg>
          Tokens
        </a>
      </div>
    </nav>
  </header>
  <div
    class="container"
  />
  <div
    class="fixed-top m-3"
  />
</div>
`;

exports[`<App /> Token should match snapshot 1`] = `
<div>
  <header>
    <nav
      class="mb-4 navbar navbar-expand navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
        id="navbar-home"
      >
        <img
          alt=""
          class="app-icon app-icon-lg"
          src=""
        />
        Etherbeam
      </a>
      <button
        aria-controls="navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbar-nav"
      >
        <a
          class="nav-link"
          href="/tokens"
          id="navbar-nav-tokens"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-dot-circle fa-w-16 fa-lg mr-1"
            data-icon="dot-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
              fill="currentColor"
            />
          </svg>
          Tokens
        </a>
      </div>
    </nav>
  </header>
  <div
    class="container"
  />
  <div
    class="fixed-top m-3"
  />
</div>
`;

exports[`<App /> Tokens should match snapshot 1`] = `
<div>
  <header>
    <nav
      class="mb-4 navbar navbar-expand navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
        id="navbar-home"
      >
        <img
          alt=""
          class="app-icon app-icon-lg"
          src=""
        />
        Etherbeam
      </a>
      <button
        aria-controls="navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbar-nav"
      >
        <a
          class="nav-link"
          href="/tokens"
          id="navbar-nav-tokens"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-dot-circle fa-w-16 fa-lg mr-1"
            data-icon="dot-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
              fill="currentColor"
            />
          </svg>
          Tokens
        </a>
      </div>
    </nav>
  </header>
  <div
    class="container"
  />
  <div
    class="fixed-top m-3"
  />
</div>
`;
